File: TCR24CS028 - Fasil Firose/src /Transition.cpp
#include "Transition.h"

Transition::Transition()
    : id(0),
      from(0),
      to(0),
      cost(0.0),
      safety(0.0),
      reliability(0.0),
      available(true) {
}

Transition::Transition(
    uint64_t id,
    uint64_t from,
    uint64_t to,
    double cost,
    double safety,
    double reliability,
    bool available
)
    : id(id),
      from(from),
      to(to),
      cost(cost),
      safety(safety),
      reliability(reliability),
      available(available) {
}

File: TCR24CS047 - Nandana Mk/transition.py
from dataclasses import dataclass


@dataclass
class Transition:
    """
    Represents a directed edge between two states.
    """

    from_state: str
    to_state: str
    cost: float
    reliability: float
    safety: float
    available: bool = True

    def __repr__(self):
        return (
            f"Transition({self.from_state} -> {self.to_state}, "
            f"Cost={self.cost}, "
            f"Reliability={self.reliability}, "
            f"Safety={self.safety}, "
            f"Available={self.available})"
        )